### ISSUE TITLE
[iOS] Undefined symbols for architecture armv7

### ISSUE THREAD

**Issue by pedrocruzlopez** [mapbox/mapbox-gl-native#14162]
Hi,
I'm trying to release my app that has Mapbox framework integrated, however when I start to build the "archive" for Generic iOS Device, I'm getting the following outputs error on my build log:

```
ld: warning: ignoring file /Users/user/Work/iOS/vinu-ios/Frameworks/Mapbox.framework/Mapbox, missing required architecture armv7 in file /Users/user/Work/iOS/vinu-ios/Frameworks/Mapbox.framework/Mapbox (2 slices)
Undefined symbols for architecture armv7:
  "_OBJC_CLASS_$_MGLAnnotationView", referenced from:
      _OBJC_CLASS_$_VCustomAnnotationView in VCustomAnnotationView.o
  "_OBJC_CLASS_$_MGLShapeOfflineRegion", referenced from:
      objc-class-ref in VSyncEngine.o
  "_MGLOfflinePackUserInfoKeyMaximumCount", referenced from:
      -[VWineInfoLightbox offlinePackDidReceiveMaximumAllowedMapboxTiles:] in VWineInfoLightbox.o
      -[VSyncEngine offlinePackDidReceiveMaximumAllowedMapboxTiles:] in VSyncEngine.o
  "_OBJC_CLASS_$_MGLTilePyramidOfflineRegion", referenced from:
      objc-class-ref in VWineInfoLightbox.o
  "_MGLOfflinePackUserInfoKeyError", referenced from:
      -[VWineInfoLightbox offlinePackDidReceiveError:] in VWineInfoLightbox.o
      -[VSyncEngine offlinePackDidReceiveError:] in VSyncEngine.o
  "_OBJC_CLASS_$_MGLMapCamera", referenced from:
      objc-class-ref in VWineInfoLightbox.o
  "_MGLOfflinePackErrorNotification", referenced from:
      -[VWineInfoLightbox awakeFromNib] in VWineInfoLightbox.o
      -[VSyncEngine initWithDelegate:] in VSyncEngine.o
  "_MGLOfflinePackMaximumMapboxTilesReachedNotification", referenced from:
      -[VWineInfoLightbox awakeFromNib] in VWineInfoLightbox.o
      -[VSyncEngine initWithDelegate:] in VSyncEngine.o
  "_OBJC_METACLASS_$_MGLPointAnnotation", referenced from:
      _OBJC_METACLASS_$_VCustomPointAnnotation in VCustomPointAnnotation.o
  "_OBJC_CLASS_$_MGLPolygon", referenced from:
      objc-class-ref in VCoordinatesBoundsCalculator.o
  "_OBJC_CLASS_$_MGLOfflineStorage", referenced from:
      objc-class-ref in VWineInfoLightbox.o
  "_OBJC_CLASS_$_MGLMapView", referenced from:
      objc-class-ref in VWineInfoLightbox.o
  "_OBJC_METACLASS_$_MGLAnnotationView", referenced from:
      _OBJC_METACLASS_$_VCustomAnnotationView in VCustomAnnotationView.o
  "_OBJC_CLASS_$_MGLMultiPolygon", referenced from:
      objc-class-ref in VSyncEngine.o
  "_OBJC_CLASS_$_MGLAnnotationImage", referenced from:
      objc-class-ref in VWineInfoLightbox.o
  "_OBJC_CLASS_$_MGLPointAnnotation", referenced from:
      _OBJC_CLASS_$_VCustomPointAnnotation in VCustomPointAnnotation.o
      objc-class-ref in VWineInfoLightbox.o
  "_MGLOfflinePackProgressChangedNotification", referenced from:
      -[VWineInfoLightbox awakeFromNib] in VWineInfoLightbox.o
      -[VSyncEngine initWithDelegate:] in VSyncEngine.o
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
It should be noted that I have manually integrated the framework as it says Mapbox [documentation](https://www.mapbox.com/install/ios/download/) (I'd prefer to avoid using CocoaPods or Carthage).
Also, the following are configurations on my target

**Valid architectures:**
<img width="542" alt="Screen Shot 2019-03-19 at 10 51 13" src="https://user-images.githubusercontent.com/12741167/54625865-b14f5800-4a35-11e9-893f-9f20361043aa.png">

**Flags:**
<img width="603" alt="Screen Shot 2019-03-19 at 10 51 21" src="https://user-images.githubusercontent.com/12741167/54625942-d643cb00-4a35-11e9-95a7-72a2de65a8e4.png">

**Link Binary with Libraries:**
<img width="816" alt="Screen Shot 2019-03-19 at 10 51 42" src="https://user-images.githubusercontent.com/12741167/54626037-01c6b580-4a36-11e9-8017-9704b6a46490.png">

**Embedded Binaries**
<img width="710" alt="Screen Shot 2019-03-19 at 11 22 12" src="https://user-images.githubusercontent.com/12741167/54627564-4ef85680-4a39-11e9-9a8d-16008a035f50.png">

**Embed Frameworks**
<img width="805" alt="Screen Shot 2019-03-19 at 11 23 03" src="https://user-images.githubusercontent.com/12741167/54627625-6fc0ac00-4a39-11e9-8d6e-afa69a886bc4.png">

**Important note: This issues only happens when I "archive" for release the app, but when I build on the simulator everything is ok.**

Am I missing some configuration?

Thanks in advance!.

### Steps to reproduce

 1. Go to Product/Archive

### Expected behavior

Build successful

### Actual behavior

A lot of missing symbols for armv7

### Configuration

**Mapbox SDK versions:** 4.7
**iOS/macOS versions:**  mojave 10.14
**Device/simulator models:** 
**Xcode version:**  10

**Comment by julianrex** [comment]
@pedrocruzlopez do you have Mapbox.framework listed under "Embedded Binaries"? If not, please ensure you add it there too.

**Comment by pedrocruzlopez** [comment]
@julianrex Yes, I do. I have updated the post with screenshots of that configs, thanks!, but I'm afraid that is not the problem.

**Comment by pedrocruzlopez** [comment]
Well, after moving the .framework to the root directory of the project, now appears to be working. Sorry for that mistake but was weird that was working on the simulator.

**Comment by julianrex** [comment]
Good to hear you have it working.

**Comment by jbarros35** [comment]
I'm using Cocoapods, its not needed to embed anything at all

**Comment by NiYun** [comment]
I tried to compile a project integrated with AdjustSDK exported by Unity2018 on Xcode 10.1, and the following error occurred:

Undefined symbols for architecture armv7:
  "_OBJC_CLASS_$_AdjustTestOptions", referenced from:
      objc-class-ref in AdjustUnity.o
  "_OBJC_CLASS_$_Adjust", referenced from:
      objc-class-ref in AdjustUnity.o
     (maybe you meant: _OBJC_CLASS_$_AdjustUnityDelegate, _OBJC_CLASS_$_AdjustUnity )
  "_OBJC_CLASS_$_ADJConfig", referenced from:
      objc-class-ref in AdjustUnity.o
  "_IOS_SendMail", referenced from:
      _PhoneHelper_IOS_SendMail_m68996C6A2D690CC7B9B3E8F1D3989392BA8F5961 in Bulk_Assembly-CSharp_16.o
     (maybe you meant: _PhoneHelper_IOS_SendMail_m68996C6A2D690CC7B9B3E8F1D3989392BA8F5961)
  "_IOS_GetUUIDString", referenced from:
      _PhoneHelper_F_GetUUID_m850D03B43001E3F8A0E8D8A9D783DC8C70B1882C in Bulk_Assembly-CSharp_16.o
      _PhoneHelper_IOS_GetUUIDString_mD1D052482047CFCCEB612ADB84384570A1C0377B in Bulk_Assembly-CSharp_16.o
     (maybe you meant: _PhoneHelper_IOS_GetUUIDString_mD1D052482047CFCCEB612ADB84384570A1C0377B)
  "_IOS_BuyItem", referenced from:
      _PhoneHelper_F_BuyItem_mE76F4E11FBAE8FB08484FE30257093157FC88627 in Bulk_Assembly-CSharp_16.o
      _PhoneHelper_IOS_BuyItem_m010DE30EF117FA91AACCD1710008DB638F1A7CDF in Bulk_Assembly-CSharp_16.o
     (maybe you meant: _PhoneHelper_IOS_BuyItem_m010DE30EF117FA91AACCD1710008DB638F1A7CDF)
  "_OBJC_CLASS_$_ADJEvent", referenced from:
      objc-class-ref in AdjustUnity.o
  "_IOS_InitShopItem", referenced from:
      _NotEnoughCoinBox_pre_Start_m333A8EF8B9E241BF54E7579170DA0FA006C3F2EE in Bulk_Assembly-CSharp_16.o
      _PhoneHelper_F_InitShopItem_mF4015BEF4C9433EF6A18358938F0828DA3808924 in Bulk_Assembly-CSharp_16.o
      _PhoneHelper_IOS_InitShopItem_m322096FF8F5E8F63395B95BFE281DDA2D8263C4D in Bulk_Assembly-CSharp_16.o
     (maybe you meant: _PhoneHelper_IOS_InitShopItem_m322096FF8F5E8F63395B95BFE281DDA2D8263C4D)
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)


Compilation error message points to Adjust, what settings need to be done?

**Comment by vijaykholiya** [comment]
I had the same issue for the framework Twilio Chat client. And when I went into carthage folder and inside it open the Info.plist file for Twilio Chat client framework the minimum deployment target was set to 11.0 but I was developing application for 10.0 so it was missing those architectures.

How I resolved: I downgraded my Twilio Chat Client framework to earlier versions where deployment target was set to 9.0 and boom it works.
